pennylane/transforms/hamiltonian_expand.py

```diff
@@ -93,6 +88,27 @@ def hamiltonian_expand(tape, group=True):
         # split H into observables [qml.PauliZ(0)], [qml.PauliX(1)] and [qml.PauliX(0)]
         tapes, fn = qml.transforms.hamiltonian_expand(tape, group=False)
         print(len(tapes)) # 3
+
+    Alternatively, if the Hamiltonian has already computed groups, they are used even if ``group=False``:
```

Comment:
Yes, I know. This is a bit of a compromise to not change argument names - it would be better to rename it to "force_grouping" or so. 

Basically, this transform is used deep inside the QNode and we want a user to control its behaviour without feeding the argument all the way to the surface. The way this is done is via the Hamiltonian... and there is never a reason _not_ to use grouping information if it's available, so ignoring the info in the Hamiltonian even if `group=False` is always suboptimal.

Reply:
> and there is never a reason not to use grouping information if it's available,

For the most part I agree but there could be characterization / error mitigation protocols where you need to investigate things at the level of granularity of single observables. Even with the documentation I find this a bit unintuitive, I'm in favour of naming it something more descriptive like `force_grouping` as you suggest, just so it's less ambiguous what it's doing.